common/store.go

```diff
@@ -61,8 +61,17 @@ func StringToBackendType(s string) BackendType {
 type Store interface {
 	// Backend returns the backend type provider of the store.
 	BackendType() BackendType
-	// Verify verifies the given key-value pair.
-	Verify(ctx context.Context, key []byte, value []byte) error
+}
+
+type VerifyArgs struct {
+	// L1 block number at which the rollup batch was submitted to the batcher inbox.
+	// This is optional, and should be set to -1 to mean to not verify the reference block number distance check.
+	//
+	// Used to determine the validity of the eigenDA batch.
+	// The eigenDA batch header contains a reference block number (RBN) which is used to pin the stake of the eigenda operators at that specific blocks.
+	// The rollup batch containing the eigenDA cert is only valid if it was included within a certain number of blocks after the RBN.
+	// validity condition is: RBN < l1_inclusion_block_number < RBN + some_delta
+	RollupL1InclusionBlockNum int64
```

Comment:
shouldn't this be `uint64`?

Reply:
I'm using -1 to mean that this query param was not passed, and hence to skip the test. Could using a pointer to uint64 instead and use nil to mean this? I don't like either tbh... but go is dumb.

Reply:
Can't you use `0` to indicate "skip the test"?

`RollupL1InclusionBlockNum` must be strictly greater RBN, and the min value of RBN is `0`, therefore `RollupL1InclusionBlockNum == 0` isn't sensical, and is free realestate to have special meaning

Reply:
> but I think its technically possible, especially on a devnet

I don't see how this would be possible

The eigenDA cert directly contains the RBN, so you can't honestly construct the cert until the reference block exists. And if the RB already exists, how could the eigenDA cert be put on chain any earlier than `RBN+1`?

Reply:
Started new PR to rebase cleanly: https://github.com/Layr-Labs/eigenda-proxy/pull/378

updated in: [2eddc29](https://github.com/Layr-Labs/eigenda-proxy/pull/378/commits/2eddc29e6489ff1e920f15a8fdc4c48325fe5474)